### PR TITLE
Adds aws collector

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Towards https://github.com/giantswarm/giantswarm/issues/2745
Towards https://github.com/giantswarm/giantswarm/issues/2888

This PR introduces a Prometheus Collector for fetching information about resources on AWS - currently just VPC info. This is towards better alerting for orphaned resources and clusters.

Next PR I will introduce a function for exposing the information we need to alert on duplicated masters.

e.g:
```
aws_operator_vpc_info
```
gives

Element | Value
-- | --
aws_operator_vpc_info{app="aws-operator",cidr="10.1.74.0/24",cluster_id="gauss",cluster_type="host",exported_cluster_id="3meei",id="vpc-0c14c2ca44ca8b438",installation="gauss",instance="192.168.96.44:8000",job="host-cluster-gauss-workload",name="3meei",namespace="giantswarm",organization="giantswarm",stack_name="cluster-3meei-guest-main",state="available"} | 1
aws_operator_vpc_info{app="aws-operator",cidr="10.1.76.0/24",cluster_id="gauss",cluster_type="host",exported_cluster_id="i3am7",id="vpc-0ed93f03539d3ac56",installation="gauss",instance="192.168.96.44:8000",job="host-cluster-gauss-workload",name="i3am7",namespace="giantswarm",organization="giantswarm",stack_name="cluster-i3am7-guest-main",state="available"} | 1
aws_operator_vpc_info{app="aws-operator",cidr="172.31.0.0/16",cluster_id="gauss",cluster_type="host",id="vpc-b851f5d3",installation="",instance="192.168.96.44:8000",job="host-cluster-gauss-workload",name="",namespace="giantswarm",organization="",stack_name="",state="available"} | 1

### CODE REVIEW

File: service/collector/collector.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+func New(config Config) (*Collector, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+
+	var emptyAwsConfig awsutil.Config
+	if config.AwsConfig == emptyAwsConfig {
+		return nil, microerror.Maskf(invalidConfigError, "config.AwsConfig must not be empty")
```
Comment (xh3b4sd) [inline service/collector/collector.go]: We do this now to get more type information when shit hits the fan. 

```
return nil, microerror.Maskf(invalidConfigError, "%T.AwsConfig must not be empty", config)
```

File: service/collector/vpc.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+			OrganizationLabel,
+			StackNameLabel,
+			StateLabel,
+		},
+		nil,
+	)
+)
+
+func (c *Collector) collectVPCs(ch chan<- prometheus.Metric) {
+	c.logger.Log("debug", "collecting metrics for vpcs")
```
Comment (xh3b4sd) [inline service/collector/vpc.go]: We do this now to be better off with the EFK magic. 

```
c.logger.Log("level", "debug", "message", "collecting metrics for vpcs")
```

File: service/collector/vpc.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+	resp, err := c.awsClients.EC2.DescribeVpcs(&ec2.DescribeVpcsInput{})
```
Comment (xh3b4sd) [inline service/collector/vpc.go]: Some caching would actually be cool so we fetch the information all the time and have a prepared bucket of information we can just send out all the time.

File: service/collector/vpc.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	if err != nil {
+		c.logger.Log("debug", "could not list vpcs: ", err)
```
Comment (xh3b4sd) [inline service/collector/vpc.go]: This should be log level error with stack information. 

```
c.logger.Log("level", "error", "message", "could not list vpcs", "stack", fmt.Sprintf("%#v", err))
```

**Comment by JosephSalisbury** [comment]
@xh3b4sd ptala

File: service/collector/vpc.go [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+		nil,
+	)
+)
+
+func (c *Collector) collectVPCs(ch chan<- prometheus.Metric) {
+	c.logger.Log("level", "debug", "message", "collecting metrics for vpcs")
+
+	resp, err := c.awsClients.EC2.DescribeVpcs(&ec2.DescribeVpcsInput{})
+	if err != nil {
+		c.logger.Log("level", "error", "message", "could not list vpcs", "error", fmt.Sprintf("%#v", err))
```
Comment (xh3b4sd) [inline service/collector/vpc.go]: The `stack` field is already used in other projects. Making up your own thing now will not help people debugging via EFK. I do not have strong feelings about the identifier here. The important thing for me is that we use the same identifier for one thing everywhere to make our lives easier. Feel free to change this everywhere though, but changing this in one project and keeping the others unchanged is not going to work that well for humans.

Comment (JosephSalisbury) [reply service/collector/vpc.go]: For whoever comes later, see https://github.com/giantswarm/operatorkit/blob/c1a1a0c6c139450e5db7c92e92fb8706984a0cf9/framework/resource/retryresource/crud_resource_ops_wrapper.go#L71 for example

File: service/collector/vpc.go [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+	for _, vpc := range resp.Vpcs {
+		cluster := ""
+		installation := ""
+		name := ""
+		organization := ""
+		stackName := ""
+		for _, tag := range vpc.Tags {
+			if *tag.Key == ClusterTag {
+				cluster = *tag.Value
+			}
```
Comment (rossf7) [inline service/collector/vpc.go]: <3 that the resource tags are exposed with the metrics.

**Pull request merged**